index/scorch/segment/zap/new_test.go

```diff
@@ -0,0 +1,77 @@
+//  Copyright (c) 2018 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zap
+
+import (
+	"testing"
+
+	"github.com/RoaringBitmap/roaring"
+)
+
+func BenchmarkRoaringAdd1(b *testing.B) {
+	benchmarkRoaringAdd(b, 1)
+}
+
+func BenchmarkRoaringAdd10(b *testing.B) {
+	benchmarkRoaringAdd(b, 10)
+}
+
+func BenchmarkRoaringAdd1K(b *testing.B) {
+	benchmarkRoaringAdd(b, 1000)
+}
+
+func BenchmarkRoaringAdd1M(b *testing.B) {
+	benchmarkRoaringAdd(b, 1000000)
+}
+
+func BenchmarkRoaringAdd10M(b *testing.B) {
+	benchmarkRoaringAdd(b, 10000000)
+}
+
+func benchmarkRoaringAdd(b *testing.B, n uint32) {
+	r := roaring.New()
+	for i := 0; i < b.N; i++ {
+		for j := uint32(0); j < n; j++ {
+			r.Add(j)
+		}
+	}
+}
+
+func BenchmarkRoaringAddRange1(b *testing.B) {
+	benchmarkRoaringAddRange(b, 1)
+}
+
+func BenchmarkRoaringAddRange10(b *testing.B) {
+	benchmarkRoaringAddRange(b, 10)
+}
+
+func BenchmarkRoaringAddRange1K(b *testing.B) {
+	benchmarkRoaringAddRange(b, 1000)
+}
+
+func BenchmarkRoaringAddRange1M(b *testing.B) {
+	benchmarkRoaringAddRange(b, 1000000)
+}
+
+func BenchmarkRoaringAddRange10M(b *testing.B) {
+	benchmarkRoaringAddRange(b, 10000000)
+}
+
+func benchmarkRoaringAddRange(b *testing.B, n uint64) {
+	r := roaring.New()
+	for i := 0; i < b.N; i++ {
+		r.AddRange(0, n)
```

Comment:
hi @steveyen , correct me if I am myopic about something obvious here. :)
May be we should benchmark , r.Add Vs r.incorporateLastRange() api for  cases like,
1. sequential docNumbers,  2. equally distant (+5 or +10) docNumbers,   3. random distributed docNumbers given that a partition may not have all docNumbers in sequence even if the fields are low cardinal. 
This may give a better insight into the actual perf gain in a real use case? I feel the benchmarking done here is skewed one?

Reply:
will do.  Just to share the thinking here, the microbenchmarks were the first thing I wrote to experiment with the idea on whether AddRange() was interesting or not to consider using, at least for sequential docNums.  But, I should add some more of these now that incorporateLastRange() exists.